metaflow/runtime.py

```diff
@@ -238,8 +245,28 @@ def execute(self):
                                         'by the end of flow.')
 
     def _killall(self):
+        # If we are here, all children have received a signal and are shutting down.
+        # We want to give them an opportunity to do so and then kill
+        live_workers = []
+        temp_workers = {}
         for worker in self._workers.values():
-            worker.kill()
+            temp_workers[worker] = 1
+        live_workers = temp_workers.keys()
+        now = int(time.time())
+        self._logger('Terminating %d active tasks...' % len(live_workers),
+                     system_msg=True, bad=True)
+        while live_workers and int(time.time()) - now < 5:
+            # While not all workers are dead and we have waited less than 5 seconds
+            live_workers = [worker for worker in live_workers if not worker.clean()]
+        if live_workers:
+            self._logger('Killing remaining %d tasks after having waited for %d seconds -- '
+                         'some tasks may not exit cleanly' % (len(live_workers),
+                                                              int(time.time()) - now),
+                         system_msg=True, bad=True)
+            for worker in live_workers:
+                worker.kill()
+        self._logger('Flushing logs -- you may see messages from the tasks that were shutdown...',
```

Comment:
Should we remove this logging message?

Reply:
I wanted to keep it to help users understand why they see printout after things have been waited for and terminated.

Reply:
I am worried that this log is too verbose. Maybe just? - `Flushing logs...`